src/pip/_internal/operations/install/wheel.py

```diff
@@ -90,14 +90,13 @@ def fix_script(path: str) -> bool:
     assert os.path.isfile(path)
 
     with open(path, "rb") as script:
-        firstline = script.readline()
-        if not firstline.startswith(b"#!python"):
+        prelude = script.readline()
+        if (m := re.match(rb"^#!python[^\s]*(\s.*)?$", prelude)) is None:
             return False
-        exename = sys.executable.encode(sys.getfilesystemencoding())
-        firstline = b"#!" + exename + os.linesep.encode("ascii")
+        prelude = ScriptMaker(None, None)._get_shebang("utf-8", m.group(1) or b"")
```

Comment:
Actually, `_get_shebang` does the wrong thing here. It doesn't correctly rewrite `#!pythonw` as the GUI version of the interpreter, if pip is being run under the CLI version. Entry points use the `ScriptMaker.make(..., {'gui': True})` mechanism to do that.

I'd suggest that we need a test here that ensures that `#!pythonw` gets rewritten as `Path(sys.executable).parent / (Path(sys.executable).stem + "w" + Path(sys.executable).suffix)`, as that's the correct behaviour according to the spec. The code as given would fail that test.